### PR TITLE
initial perms middleware work with org id filtering

### PR DESCRIPTION
**Pull request by dheniges**
Adds multiple permissions middlewares, one for generating reusable CRUD permissions, and a lower level perms middleware that can request permission to do anything.
A permission request involves two objects:
- requestorInfo
- requestedAction

With those two pieces of information, we should be able to validate any type of future permission, coming from any requester (user, API token, other). We'll have to build out the actual checking of permissions, but it won't impact express middleware or route handlers.

the BusinessUser and BusinessProject now automatically filter by org id (and require an org id)

### CODE REVIEW

File: api/src/businesstime/project.js [hunk 1]
```diff
@@ -9,14 +9,15 @@ async function create(body) {
   return createdProject.get({ plain: true })
 }
 
-async function findAll(id) {
+async function findAll(orgId) {
+  console.info({ orgId })
```
Comment (jjidt) [inline api/src/businesstime/project.js]: leftover logging

File: api/src/businesstime/user.js [hunk 2]
```diff
@@ -20,14 +20,18 @@ async function updateByEmail(email, body) {
 }
 
 // TODO convert to find all users within organization
```
Comment (jjidt) [inline api/src/businesstime/user.js]: can remove this TODO now

File: api/src/controllers/loader.js [hunk 3]
```diff
@@ -19,7 +20,7 @@ const UNAUTHENTICATED_CONTROLLERS = {
 }
 
 const loadControllers = (router, controllers, middleware) => {
-  Object.keys(controllers).forEach((path) => {
+  Object.keys(controllers).forEach(path => {
```
Comment (jjidt) [inline api/src/controllers/loader.js]: why'd the linter have you remove the parens in this file, shouldn't they be there?

File: api/src/controllers/projects.js [hunk 4]
```diff
@@ -57,6 +66,8 @@ const replaceProject = async function(req, res) {
   const { id } = req.params
   const { body } = req
 
+  // TODO: don't let users change orgId
```
Comment (jjidt) [inline api/src/controllers/projects.js:69]: is this referring to what you did on line 54? or is it in reference to changing the orgId on the actual user object?

File: api/src/libs/middleware/reqPermissionsMiddleware.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  if (typeof actionBuilder !== 'function') {
+    throw new Error('actionBuilder arg must be a function')
+  }
+  return async (req, res, next) => {
+    try {
+      const hasPermission = await checkPermission(req.requestorInfo, actionBuilder(req))
+      if (hasPermission) {
+        next()
+      } else {
+        // TODO: convert to ono?
```
Comment (jjidt) [inline api/src/libs/middleware/reqPermissionsMiddleware.js]: I think the normal error here is fine, probably only need ono if it's a rethrow or need to pass additional status code info, in this case we're just triggering a 403 for everything

**Review by jjidt — CHANGES_REQUESTED** [review]
This looks and works great, super easy to follow

Just a few cleanup comments

File: api/src/controllers/projects.js (hunk 4 above)
Comment (dheniges) [reply api/src/controllers/projects.js:69]: it's in reference to not being able to change an org id on the project, like what I did on the users. I backlogged an asana item to whitelist update fields, so I'll remove the TODO then

File: api/src/controllers/loader.js (hunk 3 above)
Comment (dheniges) [reply api/src/controllers/loader.js]: I don't know. I re-saved it now and it converted it back. I didn't manually make the change. Le sigh.

**Comment by dheniges** [comment]
@jjidt ready for re-review (I re-requested it, but not sure how those notifications work)

**Review by jjidt — APPROVED** [review]

**Pull request merged**